pyOxygenSCPI/oxygenscpi.py

```diff
@@ -532,31 +579,162 @@ def setElogPeriod(self, period):
+        # datetime string and the remaining values are strings that can
+        # be converted to floating point numbers.
+        if self.elogTimestamp == "ABS":
+            new_array = []
+            dtime = data_array[0].replace('"', '')
+            # Oxygen's datetime with whole seconds does not display microseconds
+            # for those cases the ".%f" formatting is excluded.
+            fmt = '%Y-%m-%dT%H:%M:%S.%f' if '.' in dtime else '%Y-%m-%dT%H:%M:%S'
+            new_array.append(dt.datetime.strptime(dtime, fmt))
+            new_array.extend(float(value) for value in data_array[1:])
+        else:
+            new_array = [float(value) for value in data_array]
+        return new_array
+
+    def fetchElogAccumulated(self, timeout=10):
+        """Fetch ELOG until the actual timestamp is reached.
+
+        This function blocks the execution and keeps fetching elog values until
+        a fetched timestamp is higher than the timestamp saved at the moment in
+        which the function was called. If called right after starting the
+        external logging, it is possible that the function needs longer because
+        the update of the Dewetron's buffer is not instantaneous.
+
+        Depending on the internet connection, the function itself can take a few
+        seconds to be excecuted.
+        
+        It requires the elog timestamp to be either 'ABS' or 'ELOG'.
+        - With 'ABS' timestamp, the stop condition will compare the absolute
+        timestamp from the system executing the function with the timestamp from
+        the operating system in which Oxygen is running.
+        - With 'ELOG' timestamp, the stop condition will be met when the fetched
+        timestamp added to the timestamp in which the startElog function was
+        called (tracked by _localElogStartTime) attribute is higher than the
+        timestamp from execution of the function.
+
+        Args:
+            timeout (float): timeout for the accumulated fetching.
+
+        Returns:
+            List of lists (matrix like) containing the accumulated fetched values
+            converted to float (datetime for values at first column if timestamp
+            is ABS.)
+        """
+
+        call_tstamp = dt.datetime.now()
+
+        def stopCondition(tstamp):
+            """Checks if the measured timestamp has reached the call timestamp.
+            """
+            # Case for ELOG timestamp
+            if self.elogTimestamp == "ELOG":
+                tstamp = float(tstamp)
+                return self._localElogStartTime + dt.timedelta(seconds=tstamp) >= call_tstamp
+            # Case for ABS timestmap
+            tstamp = tstamp.replace('"','')
+            tstamp = dt.datetime.strptime(tstamp, '%Y-%m-%dT%H:%M:%S.%f')
+            return tstamp >= call_tstamp
+
+        # This function works only for ELOG and ABS timestamps
+        if self.elogTimestamp in ("ELOG", "ABS"):
```

Comment:
We should raise an exception if unsupported timestamps are in use, otherwise we cannot distinguish between timeouts or wrong use

Reply:
I included the raise of Exception for the invalid timestamps in a commit. I set it to be a general `Exception` because the module still has no custom exceptions.

Regarding this topic, I would suggest creating a custom Exception called `DewetronError`, which whenever raised, would call the function `getErrorSingle` or `getErrorAll`, and print the error read from SCPI. It might also be interesting to keep track of the errors in a queued attribute. I can implement that later if you agree.

Reply:
I agree, a more descriptive Exception may be suitable here. But this should be done in a larger refactoring step to be consistent across all functions. Since there is now more interest in this library (thank you!) I am planning to add a unit test framework to keep this library stable and working across future changes - and refactoring/better documenting it while doing so. You are always welcome to add suggestions & PRs.